### PR TITLE
Bluetooth: Mesh: Add new Sensor API support to Light Ctrl shell

### PR DESCRIPTION
**Pull request by akredalen**
* Light Ctrl shell can be used in an application with new sensor API.
* Small fix: updated shell_ctrl_cli.c to comply with split model structure.

### CODE REVIEW

**Review by ludvigsj — APPROVED** [review]

**Review by HaavardRei — APPROVED** [review]

File: subsys/bluetooth/mesh/shell/shell_ctrl_cli.c [hunk 1]
```diff
@@ -27,7 +27,7 @@ static int cmd_mode_get(const struct shell *shell, size_t argc, char *argv[])
 		return -ENODEV;
 	}
 
-	struct bt_mesh_light_ctrl_cli *cli = mod->user_data;
```
Comment (alxelax) [inline subsys/bluetooth/mesh/shell/shell_ctrl_cli.c:30]: This is not a unique light control client issue. I took a quick look over shell clients, looks, this is a general problem. Currently, almost all clients cannot be built. I'm not sure if it should be fixed in this PR, but obviously should.

**Review by Andrewpini — APPROVED** [review]

**Review by alxelax — COMMENTED** [review]
LGTM, but I have question. Will it be built with `BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE=y`? I guess, it should be while we support legacy API,

File: subsys/bluetooth/mesh/shell/shell_ctrl_cli.c (hunk 1 above)
Comment (alxelax) [reply subsys/bluetooth/mesh/shell/shell_ctrl_cli.c:30]: Created JIRA ticket for this. Let's do it in another PR.

**Comment by akredalen** [comment]
> LGTM, but I have question. Will it be built with `BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE=y`? I guess, it should be while we support legacy sensor API.

Yes, the light ctrl shell was tested in a custom sample with and without the legacy API enabled.

File: subsys/bluetooth/mesh/shell/shell_ctrl_cli.c (hunk 1 above)
Comment (akredalen) [reply subsys/bluetooth/mesh/shell/shell_ctrl_cli.c:30]: I will remove this part from this PR then.

Comment (alxelax) [reply subsys/bluetooth/mesh/shell/shell_ctrl_cli.c:30]: No, please, leave it as is. Otherwise, the client cannot be built

**Review by alxelax — APPROVED** [review]

**Pull request merged**